### PR TITLE
Support OpenID discovery for RequestAuthentication

### PR DESCRIPTION
**Pull request by diemtvu**
Issue:
https://github.com/istio/istio/issues/21954

### CODE REVIEW

File: pilot/pkg/model/authentication.go [hunk 1]
```diff
@@ -105,6 +105,8 @@ func initAuthenticationPolicies(env *Environment) (*AuthenticationPolicies, erro
 
 func (policy *AuthenticationPolicies) addRequestAuthentication(configs []Config) {
 	for _, config := range configs {
+		reqPolicy := config.Spec.(*v1beta1.RequestAuthentication)
```
Comment (yangminzhu) [inline pilot/pkg/model/authentication.go]: nit: add a comment about why we need these.

**Comment by diemtvu** [comment]
> Do you plan to add some e2e tests? Maybe in a later PR.

I'm still thinking how to do it in e2e. We need a real public URI, with proper content follow OpenID discovery spec. I'm not sure if put the files in github can serve the purpose, so may be in later PR.

File: pilot/pkg/model/authentication.go (hunk 1 above)
Comment (diemtvu) [reply pilot/pkg/model/authentication.go]: done

**Comment by diemtvu** [comment]
@rshriram , can you review on behalf of Networking WG. Tks

**Pull request merged**